src/Storage/Mysql/UserRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Bugcache\Storage\Mysql;
+
+use Amp\Mysql\Pool;
+use Amp\Mysql\ResultSet;
+use Amp\Promise;
+use Bugcache\Storage;
+use Generator;
+use function Amp\resolve;
+
+class UserRepository implements Storage\UserRepository {
+    private $mysql;
+
+    public function __construct(Pool $mysql) {
+        $this->mysql = $mysql;
+    }
+
+    public function findByName(string $username): Promise {
+        $sql = "SELECT id, name FROM users WHERE name = ?";
+
+        return resolve($this->fetchUser($sql, [$username]));
+    }
+
+    private function fetchUser(string $sql, array $params = []): Generator {
+        /** @var ResultSet $result */
+        $result = yield $this->mysql->prepare($sql, $params);
```

Comment:
The use case where they are executed multiple times at the same time should work flawlessly?

Also, we're anyway tying everything to certain Amp libraries, then we can just as well tie to Aerys. But we possibly might want to have our own interface specifying startup() method or similar?